engine/src/configuration/configuration.h

```diff
@@ -0,0 +1,55 @@
+#ifndef CONFIGURATION_H
+#define CONFIGURATION_H
+
+
+struct Configuration
+{
+    // Config Structs Declaration
+
+    // These are settings that are relevant to collision.cpp and are marked as hacks
+    // or their comments are unclear or they are unclear
+    struct CollisionHacks {
+        const float collision_hack_distance;
+        const bool collision_damage_to_ai;
+        const bool crash_dock_hangar;
+        const bool crash_dock_unit;
+        const float front_collision_hack_angle;
+        const float front_collision_hack_distance;
+        const bool cargo_deals_collide_damage;
+        //const int upgrade_faction = (cargo_deals_collide_damage ? -1 : FactionUtil::GetUpgradeFaction())
+
+        CollisionHacks();
+    };
+
+    struct Logging {
+        const bool verbose_debug;
+        Logging();
+    } logging;
+
+    struct Physics {
+        const float collision_scale_factor;
+        const float inelastic_scale;
+        const float kilojoules_per_damage;
+        const float max_force_multiplier;
+        const float max_torque_multiplier;
+        const float minimum_mass;
+        const float minimum_time;             // between recorded player collisions
+
+        // Disabled bouncing missile option. Missiles always explode when colliding with something.
+        //static bool does_missile_bounce = GameConfig::GetVariable( "physics", "missile_bounce", false);
+
+        Physics();
+    } physics;
+
+    struct Weapons {
+        const bool can_fire_in_cloak;
+        const bool can_fire_in_spec;
+        Weapons();
+    } weapons;
```

Comment:
It would be better to declare these in a namespace (Vegastrike::Config), outside the class object, and then reference it.

Reply:
Do you mean split the struct definition and move it outside and then only have X x; inside the class?

Reply:
If this is the case, I'm not sure I see the value. 
It's in a header file and we don't want to expose it, so we use a namespace. 
Why not just place it within the struct under private? 
Or keep it the way it is (anonymous structs). After all, we don't foresee any other instance of these inner structs.

This really is a question - what is the value of changing this?

Reply:
It helps with unit testing for one.

Secondly its easier to read.

Honestly I don't see an issue exposing it too.

If we want to keep it private then we can stick it into an internal only header and do PIMPL...but for the moment that's too much work.

Personally I don't care for declaring structures and classes within other structures and classes. It's messy.

I also don't like a anonymous types, makes unit testing harder.

That's my opinion; go with what our code standard says as far actual usage as it may differ from my opinion.

Reply:
Resolving this in favor of continuing the discussion in https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/484#pullrequestreview-700156858 as this is now referencing out-of-date code